packages/web-react/src/hooks/useCollapsible.ts

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react';
+import { ClickEvent, useCollapsibleProperties, useCollapsibleReturn } from '../types';
+
+export const useCollapsible = (props: useCollapsibleProperties) => {
+  const {
+    id,
+    collapsed,
+    contentReference,
+    responsive,
+    UNSAFE_style,
+    UNSAFE_className,
+    contentProps,
+    wrapperClassName,
+    contentClassName,
+    hideOnCollapse,
+    ...rest
+  } = props;
+
+  const INSTANCE_ID = id || `collapsible_${(Math.random() + 1).toString(36).substring(7)}`;
+  const BREAKPOINT_MOBILE = 0;
+  const BREAKPOINT_TABLET = 768;
+  const BREAKPOINT_DESKTOP = 1280;
+  const NAME_ARIA_EXPANDED = 'aria-expanded';
+  const NAME_ARIA_CONTROLS = 'aria-controls';
+  const CLASSNAME_EXPANDED = 'is-expanded';
+  const CLASSNAME_COLLAPSED = 'is-collapsed';
+
+  const [isCollapsed, setCollapsed] = useState(collapsed);
+  const [height, setHeight] = useState<number | 'initial'>(0);
+  const [triggered, setTriggered] = useState(false);
+
+  const isResponsive = () => {
```

Comment:
I am afraid that the responsiveness should be controlled through the web package. We presume that every product can have its own breakpoints and we are not able to pass the design tokens to JavaScript right now.